### PR TITLE
Issue 15 - Endpoint to get an event

### LINKED ISSUE #15: Create an Express API endpoint for getting an Event
As a user I want to be able to look at the contents of any event given an event ID.

Todo:

- Take a look at this document where it mentions getting the event for a specific ID. https://gcurealestate.atlassian.net/wiki/spaces/SD/pages/2719745/REST+API+for+backend

- This is a GET route that we will be implementing. Look to the already existing userFetch.ts file in the repository to get an idea of what this will look like.

- The route should return a 200 response on success, or an error if unsuccessful (400 responses if the user provides an invalid ID, 500 if something fails on the server side, etc).

- should return a JSON object containing all the event data.

- Remember to reach out if you get stuck. Or ask ChatGPT. Just don't get stuck long term, find someone or something to help you get unstuck.

Deliverables:

- a GET /event api route which when passed the ID will return the event that ID is associated with.

- Remember to test your changes on your local machine, ideally attach a screenshot showing you hitting the endpoint and getting the event data, as well as a screenshot of the error handling responses. You can attach those in the comments on this issue. Once done push to test and check that your changes still work there.

### PR DESCRIPTION
**Pull request by Blake0206**
Closes #15 

**Reached through** _**/api/v1/events/{eventID}**_

**Example response**
Screenshot 2024-11-19 at 8 14 12 PM

**If the event id is not found**
Screenshot 2024-11-19 at 8 14 35 PM

### CODE REVIEW

File: backend/src/routes/api/v1/events/getAllEvents.ts
Comment (ItsRiprod) [inline backend/src/routes/api/v1/events/getAllEvents.ts:1]: Nice - good catch on the error instanceof Error - not a lot of people catch that

File: backend/src/routes/api/v1/user/userFetch.ts
Comment (ItsRiprod) [inline backend/src/routes/api/v1/user/userFetch.ts:1]: My one comment is its a good idea to put all your types into a types/ folder 

So backend/types/userTypes.ts with the type in it

Then import { UserData } from '../../../types/userTypes' 

This way the UserData object can be used anywhere and can also be easily copied to the front end to ensure we're using the same type!

**Review by ItsRiprod — APPROVED** [review]
Overall looks good!

File: backend/src/routes/types/eventTypes.ts
Comment (friesentyler) [inline backend/src/routes/types/eventTypes.ts:1]: Nice, this will be helpful for the other people working on Event and User related endpoints.

**Review by friesentyler — APPROVED** [review]

**Pull request merged**